### PR TITLE
Adds MessageRecipient component

### PR DESCRIPTION
**Pull request by webinista**
- Closes department-of-veterans-affairs/messaging-fe#36

### CODE REVIEW

File: src/js/messaging/reducers/compose.js [hunk 1]
```diff
@@ -1,17 +1,45 @@
 import set from 'lodash/fp/set';
-import { SET_CATEGORY, SET_SUBJECT } from '../actions/compose.js';
+
+import {
+  SET_CATEGORY,
+  SET_RECIPIENT,
+  SET_SUBJECT,
+  FETCH_RECIPIENTS_SUCCESS,
+  FETCH_RECIPIENTS_FAILURE
+} from '../actions/compose.js';
```
Comment (U-DON) [inline src/js/messaging/reducers/compose.js]: Can remove the .js suffix from imports.

File: src/js/messaging/components/compose/MessageRecipient.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { makeField } from '../../../common/model/fields.js';
+
+class MessageRecipient extends React.Component {
+  render() {
+    const recipientValue = makeField(this.props.value);
+    return (
+      <div className="messaging-recipient">
+        <div className="messaging-recipient">
+          <ErrorableSelect
+              label="Recipient:"
```
Comment (U-DON) [inline src/js/messaging/components/compose/MessageRecipient.jsx]: Not `label="To: "`?

File: src/js/messaging/actions/compose.js [hunk 3]
```diff
@@ -15,3 +20,21 @@ export function setSubject(value) {
     value
   };
 }
+
+export function setRecipient(recipient) {
```
Comment (U-DON) [inline src/js/messaging/actions/compose.js]: Maybe update the other `set*` function arguments to be consistent in style.

File: src/js/messaging/containers/Compose.jsx [hunk 4]
```diff
@@ -4,13 +4,24 @@ import { connect } from 'react-redux';
...
 import MessageCategory from '../components/compose/MessageCategory';
 import MessageSubject from '../components/compose/MessageSubject';
-import { setCategory, setSubject } from '../actions/compose.js';
+import MessageRecipient from '../components/compose/MessageRecipient';
+import {
+  setCategory,
+  setRecipient,
+  setSubject,
+  fetchRecipients
+} from '../actions/compose.js';
```
Comment (U-DON) [inline src/js/messaging/containers/Compose.jsx]: Don't need the .js suffix here (and for the config import above).

File: src/js/messaging/components/compose/MessageRecipient.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import ErrorableSelect from '../../../common/components/form-elements/ErrorableSelect';
+import { makeField } from '../../../common/model/fields.js';
+
+class MessageRecipient extends React.Component {
+  render() {
+    const recipientValue = makeField(this.props.value);
+    return (
+      <div className="messaging-recipient">
+        <div className="messaging-recipient">
```
Comment (U-DON) [inline src/js/messaging/components/compose/MessageRecipient.jsx]: Same div but nested. Can this be removed?

File: src/js/messaging/components/compose/MessageRecipient.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import ErrorableSelect from '../../../common/components/form-elements/ErrorableSelect';
+import { makeField } from '../../../common/model/fields.js';
```
Comment (U-DON) [inline src/js/messaging/components/compose/MessageRecipient.jsx]: Same about suffix in imports.

File: src/js/messaging/containers/Compose.jsx [hunk 7]
```diff
@@ -22,6 +33,10 @@ class Compose extends React.Component {
     this.props.setSubject(valueObj);
   }
 
+  dispatchRecipientChange(event) {
```
Comment (U-DON) [inline src/js/messaging/containers/Compose.jsx]: Standardize on the argument name with the other `dispatch*` functions.

Although, on that topic, I'm not sure if `dispatch*` is the right name. Like for `dispatchCategoryChange`, it's actually calling two actions that have been bound to the dispatch. Perhaps `handle*`?

File: src/js/messaging/containers/Compose.jsx [hunk 8]
```diff
@@ -30,6 +45,13 @@ class Compose extends React.Component {
...
+
+        <MessageRecipient
+            categories={messageCategories}
+            errorMessage={composeMessageErrors.recipient}
+            onValueChange={this.dispatchRecipientChange}
+            options={this.props.compose.recipients}
+            value={this.props.compose.recipient}/>
         <MessageCategory
             categories={messageCategories}
             errorMessage={composeMessageErrors.category}
```
Comment (U-DON) [inline src/js/messaging/containers/Compose.jsx]: Below, correct typo to `composeMessageErrors.subject`.

**Comment by webinista** [comment]
This will have a lot of conflicts with #3059. Going to sit on this one for a little bit until that PR is merged.

File: src/js/messaging/components/compose/MessageRecipient.jsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  categories: React.PropTypes.arrayOf(
+    React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.shape({
+        label: React.PropTypes.string,
+        value: React.PropTypes.number }),
+      React.PropTypes.shape({
+        label: React.PropTypes.string,
+        value: React.PropTypes.string }),
+    ])).isRequired
```
Comment (U-DON) [inline src/js/messaging/components/compose/MessageRecipient.jsx]: Props should be updated; most of these aren't being used, and you've added `value` and `options`.

File: src/js/messaging/components/compose/MessageCategory.jsx [hunk 10]
```diff
@@ -9,26 +8,14 @@ class MessageCategory extends React.Component {
...
-              value={categoryValue}/>
-        </div>
+      <div className="messaging-category">
+        <ErrorableSelect
+            additionalClass="messaging-category-select"
+            label="Category"
+            name="messageCategory"
+            onValueChange={this.props.onValueChange}
+            options={categories}
+            value={categoryValue}/>
```
Comment (U-DON) [inline src/js/messaging/components/compose/MessageCategory.jsx]: Update `propTypes` below.

File: src/js/messaging/components/compose/MessageSubject.jsx [hunk 11]
```diff
@@ -9,12 +9,13 @@ class MessageSubject extends React.Component {
     return (
       <div className="messaging-subject">
         <ErrorableTextInput
+            additionalClass="messaging-subject-input"
             errorMessage={this.props.errorMessage}
-            label="Subject Line:"
+            label="Subject"
```
Comment (U-DON) [inline src/js/messaging/components/compose/MessageSubject.jsx]: I still see "Subject Line: " in the latest (I think) mocks.

File: src/js/messaging/components/compose/MessageSubject.jsx [hunk 12]
```diff
@@ -26,6 +27,8 @@ MessageSubject.propTypes = {
   subjectClass: React.PropTypes.string,
   name: React.PropTypes.string,
```
Comment (U-DON) [inline src/js/messaging/components/compose/MessageSubject.jsx]: Couple of unused props.

**Comment by U-DON** [comment]
LGTM. Feel free to merge if I don't get to it after your changes.

File: src/js/messaging/components/compose/MessageSubject.jsx (hunk 11 above)
Comment (webinista) [inline src/js/messaging/components/compose/MessageSubject.jsx]: "Subject Line" is in Compose.jsx as the `legend` for the `fieldset` that wraps this and the category field. This is the label for just the subject field.

**Comment by webinista** [comment]
Leaving the `errorMessage` prop in here because I'll have to add it later for validation. The error message needs to be set conditionally, triggered by `onblur` or `onsubmit`, otherwise the component will show an error state.

Tracking error checking in department-of-veterans-affairs/messaging-fe#55

**Pull request merged**